unyt/_array_functions.py

```diff
@@ -216,8 +216,8 @@ def _validate_units_consistency(objs):
     # because it's already a necessary condition for numpy to use our
     # custom implementations
     units = get_units(objs)
-    sunits = set(units)
-    if len(sunits) == 1:
+    unique_units = set(units)
+    if len(unique_units) == 1 or all(u.is_dimensionless for u in unique_units):
```

Comment:
Could you explain a little bit more how this resolves the issue (sorry, it may be because it's late for me)?

Reply:
the problem was that 
- `Unit()` and `Unit(registry=UnitRegistry())` have different hash, so they are considered as 2 separate unique units
- `Unit()` (or `NULL_UNIT` is this context) gets automatically inserted for non-unyt array-like data structures only because we want them to be treated as dimensionless

This condition makes it so all dimensionless units are treated as equal as long as nothing else is received.